### DISCUSSION TITLE
Design Patter for MVC Select Options

### DISCUSSION THREAD

**Discussion by GreggThelen** [MarimerLLC/csla#3262] [Questions] [answered]
I've an employee object with a child collection of employee categories.
It was easy enough to load up an employee object with a child collection of employee categories that may be displayed in a select input item.

However, if the form is submitted the child collection (in the dropdown) isn't submitted with the form so when there is an error the page reload can load the employee object (and error messages) but the employee categories are no longer available and cannot reload the <select>.

How do people typically handle this?
Do you reload the child collection in the controller?
Do you simply load the child collection on the page itself?

Just looking for a good pattern here.

**Comment by GreggThelen** [comment, marked as answer]
I ended up injecting the child collection (of options) into the page.
This seems to be the cleanest approach.

If there is a better approach I'm all ears.  ;)

**Reply by rockfordlhotka** [reply]
This is to populate the dropdown with name/value pairs so the id value can be mapped into a business object property?

I'd totally inject that into the page - those really aren't children of the business object - they are UI metadata.

**Reply by GreggThelen** [reply]
Thanks, that is what I ended up doing.
It is pretty slick.
